src/sentry/consumers/__init__.py

```diff
@@ -405,6 +405,34 @@ def get_stream_processor(
             f"Invalid consumer definition configured for {consumer_name}"
         ) from e
 
+    post_process_forwarders = [
+        "post-process-forwarder-errors",
+        "post-process-forwarder-transactions",
+        "post-process-forwarder-issue-platform",
+    ]
+    subscription_result_consumers = [
+        "events-subscription-results",
+        "transactions-subscription-results",
+        "generic-metrics-subscription-results",
+        "metrics-subscription-results",
+    ]
+    consumers_that_should_have_dlq_but_dont = [
+        "process-spans",
+        "detect-performance-issues",
+        "ingest-monitors",
+        "metrics-last-seen-updater",
+        "generic-metrics-last-seen-updater",
```

Comment:
Looks like we have a lot more consumers without DLQs now, e.g. "ingest-profiles", "ingest-replay-recordings", and others. Do we need to include those too?

Generally, to avoid unexpected hard failures (e.g. if we forget to allowlist some consumer from getsentry), I'd suggest to go with the warning only first, so we can then filter in logs what we might've missed.

Reply:
Isn't the assertion going to make CI fail before we try to run anything in production anyway ?

Reply:
On second thoughts, I'll probably just rewrite this into a test and forgo this runtime stuff